pygridsim/parameters.py

```diff
@@ -51,45 +67,76 @@ def make_load_node(load_params, load_type, count):
         load object
     """
     check_valid_params(load_params, defaults.VALID_LOAD_PARAMS)
+    load_type_obj = get_enum_obj(LoadType, load_type)
 
     load : Load = altdss.Load.new('load' + str(count))
     load.Bus1 = 'load' + str(count)
     load.Phases = get_param(load_params, "phases", defaults.PHASES)
-    load.kV = get_param(load_params, "kV", random_param(load_type.value["kV"]))
-    load.kW = get_param(load_params, "kW", random_param(load_type.value["kW"]))
-    load.kvar = get_param(load_params, "kVar", random_param(load_type.value["kVar"]))
+    for attr in ["kV", "kW", "kvar"]:
+        load_type_param = LOAD_CONFIGURATIONS[load_type_obj][attr]
+        setattr(load, attr, get_param(load_params, attr, random_param(load_type_param)))
     load.Daily = 'default'
 
-    if (load.kV) < 0:
-        raise ValueError("Cannot have negative voltage in load")
     return load
 
-def make_source_node(source_params, source_type, count, num_in_batch = 1):
+def make_source_node(source_params, source_type):
     """
     Make a source node with the parmeters given, filling in with defaults for
     any undefined but required parameter. Parse through the parameters, potentially throwing errors and warnings if
-    one of the parameter names is invalid.
+    one of the parameter names is invalid. Note that this updates the source node if one already exists
 
     Args:
         source_params: any specified parameters to override default ones
-        count: how many sources have already been made, to not use repeat names
-        num_in_batch: how many to batch into this same source. note this only causes a scaled kV
-        TODO: num, once we get the transformer thing working
     Return:
         source object
-    
-    TODO: There is a whole set of other vsource properties to set, like impedance and resistance
-    https://github.com/dss-extensions/AltDSS-Python/blob/2b6fa7e5961cedaf8482c07d377b20bdab4a1bee/altdss/Vsource.py#L694
     """
     check_valid_params(source_params, defaults.VALID_SOURCE_PARAMS)
+    source_type_obj = get_enum_obj(SourceType, source_type)
 
     source = altdss.Vsource[0]
     source.Bus1 = 'source'
     source.Phases = get_param(source_params, "phases", defaults.PHASES)
-    source.BasekV = get_param(source_params, "kV", num_in_batch*random_param(source_type.value))
+    source_type_param = SOURCE_CONFIGURATIONS[source_type_obj]["kV"]
+    source.BasekV = get_param(source_params, "kV", random_param(source_type_param))
     source.Frequency = get_param(source_params, "frequency", defaults.FREQUENCY)
 
-    if (source.BasekV) < 0:
-        raise ValueError("Cannot have negative voltage in source")
+    for imp in ["R0", "R1", "X0", "X1"]:
```

Comment:
I wonder if for fixed variables such as `["R0", "R1", "X0", "X1"]` should be named in the config file such that for any configuration changes, we can refer to that file instead of having it fixed in the code?